packages/shared/src/components/notifications/PersonalizedDigest.tsx

```diff
@@ -0,0 +1,348 @@
+  const router = useRouter();
+  const { isPlus } = usePlusSubscription();
+  const { isPushSupported } = usePushNotificationContext();
+  const { user } = useAuthContext();
+  const { logEvent } = useLogContext();
+  const { openModal } = useLazyModal();
+  const briefUIFeatureValue = useFeature(briefUIFeature);
+  const {
+    getPersonalizedDigest,
+    isLoading,
+    subscribePersonalizedDigest,
+    unsubscribePersonalizedDigest,
+  } = usePersonalizedDigest();
+  const [digestTimeIndex, setDigestTimeIndex] = useState<number | undefined>(8);
+
+  const selectedDigest = useMemo(() => {
+    if (isLoading) {
+      return null;
+    }
+
+    const brief = getPersonalizedDigest(UserPersonalizedDigestType.Brief);
+
+    if (brief) {
+      return brief;
+    }
+
+    const digest = getPersonalizedDigest(UserPersonalizedDigestType.Digest);
+
+    if (digest) {
+      return digest;
+    }
+
+    return null;
+  }, [getPersonalizedDigest, isLoading]);
+
+  const onLogToggle = (isEnabled: boolean, category: NotificationCategory) => {
+    const baseLogProps = {
+      extra: JSON.stringify({ channel, category }),
+    };
+    if (isEnabled) {
+      logEvent({
+        event_name: LogEvent.EnableNotification,
+        ...baseLogProps,
+      });
+    } else {
+      logEvent({
+        event_name: LogEvent.DisableNotification,
+        ...baseLogProps,
+      });
+    }
+  };
+
+  const setCustomTime = (
+    type: UserPersonalizedDigestType,
+    preferredHour: number,
+    setHour: React.Dispatch<SetStateAction<number | undefined>>,
+  ): void => {
+    if (type === UserPersonalizedDigestType.ReadingReminder) {
+      logEvent({
+        event_name: LogEvent.ScheduleReadingReminder,
+        extra: JSON.stringify({
+          hour: preferredHour,
+          timezone: user?.timezone,
+        }),
+      });
+    } else if (type === UserPersonalizedDigestType.Digest) {
+      logEvent({
+        event_name: LogEvent.ScheduleDigest,
+        extra: JSON.stringify({
+          hour: preferredHour,
+          timezone: user?.timezone,
+          frequency: selectedDigest.flags.sendType,
+        }),
+      });
+    }
+
+    subscribePersonalizedDigest({
+      type,
+      hour: preferredHour,
+      sendType: selectedDigest.flags.sendType,
+      flags: selectedDigest.flags,
+    });
+    setHour(preferredHour);
+  };
+
+  const isChecked = ns?.briefing_ready?.[channel] === 'subscribed';
+
+  const onToggleBriefing = () => {
+    toggleSetting('briefing_ready', channel);
+
+    if (isChecked) {
+      unsubscribePersonalizedDigest({
+        type: selectedDigest?.type,
+      });
+    }
+  };
+
+  const onSubscribeDigest = async ({
+    type,
+    sendType,
+    flags,
+    preferredHour,
+  }: {
+    type: UserPersonalizedDigestType;
+    sendType: SendType;
+    flags?: Pick<UserPersonalizedDigest['flags'], 'email' | 'slack'>;
+    preferredHour?: number;
+  }): Promise<void> => {
+    onLogToggle(true, NotificationCategory.Digest);
```

Comment:
maybe is better to make this function accepting an object `{ enabled: boolean, category: NotificationCategory }`. Wdyt?